### PR TITLE
Happy Thought by Kartika

### PR DESCRIPTION
**Pull request by katirum**
Link to my deployed project:
https://yourhappiness.netlify.app/

### CODE REVIEW

File: code/src/components/ThoughtForm.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      <div className="post-thought">
+        <p>What&apos;s making you happy right now?</p>
+        <textarea
+          value={newThought}
+          onChange={onNewThoughtChange}
+          className="thought"
+          placeholder="write something nice here..." />
+        <p
+          className="character-counter">
+          {140 - newThought.length} / 140
```
Comment (joeohm) [inline code/src/components/ThoughtForm.js:18]: Nice solution for the character count. I had implemented this as well but this was a much simpler solution for the same functionality, so kudos!

File: code/src/components/ThoughtList.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      {thoughtList.map((thought) => (
+        <div key={thought._id} className="thought-list">
+          <h4 className="the-thought">{thought.message}</h4>
+          <div className="thought-detail">
+            <button
+              type="button"
+              /* an if conditional that says if the count of the heart is zero,
+              then the className would be empty-button,
+              otherwise it would be like-button */
+              className={thought.hearts === 0 ? 'empty-button' : 'like-button'}
```
Comment (joeohm) [inline code/src/components/ThoughtList.js:23]: It doesn't make much sense to show that the button is pressed if someone else liked the post, but the instructions weren't very clear about this and being able to like a post infinitely also makes no sense 🥴

**Review by joeohm — COMMENTED** [review]
Beautiful page with a very nice background gradient shifting effect!

- ##### Am I able to understand the code easily?
Yes very nice readable code, with complimentary comments. The App component is kept neat, and then the code is divided up into two major branches in ThoughtForm and ThoughtList for the main functionality of the app.

- ##### Does the code have lots of duplicates? Could it be tidied up?
The code does not have a lot of duplicates, and uses for example several CSS selectors for elements that share some styling

- ##### Does it work or does anything look broken?
Everything works as intended, no errors or bugs.

- ##### Does it follow the general and blue requirements?
It does! 🥳🎉

Well done on this week's project Tika!
// Joel Ö.

File: README.md [hunk 3]
```diff
@@ -1,13 +1,11 @@
...
 ## The problem
 
-Describe how you approached to problem, and what tools and techniques you used to solve it. How did you plan? What technologies did you use? If you had more time, what would be next?
+I attended the live sessions, read all the materials and watched the recorded videos to start with. I also google a lot and look at other students' project to see how they did it. I discussed many problems with my teammates for many hours.
+If I had more time, I would use it to rewatch more all the videos and maybe put more fun stuff on the page like putting in your username, but for now I'm happy with it.
 
 ## View it live
-
-Every project should be deployed somewhere. Be sure to include the link to the deployed project so that the viewer can click around and see what it's all about.
+https://sage-sable-b6a30a.netlify.app
```
[the remark this answers is not in the document]
Comment (katirum) [reply README.md]: Thank you so much Joel! Forgot about this and I've changed it recently.

File: code/src/components/ThoughtList.js (hunk 2 above)
Comment (katirum) [reply code/src/components/ThoughtList.js:23]: Haha how do you mean?

**Pull request closed without merging**